### PR TITLE
WIP update json-schema to latest (~2.6.0)

### PR DESCRIPTION
**Pull request by anthonator**

### CODE REVIEW

**Comment by anthonator** [comment]
For the remaining errors when I test them in isolation they all pass. If I run them together they fail. I'm thinking this could be caused by some artifacts being created and used across tests. Maybe `create_schema` is causing issues?

**Comment by seanpdoyle** [comment]
@anthonator this is a great start, thanks!

**Comment by anthonator** [comment]
:smile:

There definitely seems to be some weird artifact stuff happening. The tests seem to be using different schemas than expected.

For instance, if I change the schema name to something non-generic (ie `foo_schema_1) it works. Rename it back to a generic name used between tests (ie `foo_schema`) and they start failing.

**Comment by anthonator** [comment]
This might be a caching issue on `json-schema`'s end.

https://github.com/ruby-json-schema/json-schema/issues/310

**Comment by anthonator** [comment]
Yep, definitely a caching issue. I'm clearing `json-schema`'s cache before creating a new schema.

I haven't thought too deeply about this. Further discussion around the appropriateness of caching within `json_matchers` may be necessary.

**Comment by seanpdoyle** [comment]
@anthonator CI is green, so something is working :+1:.

File: spec/support/file_helpers.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@ module FileHelpers
   ORIGINAL_SCHEMA_ROOT = JsonMatchers.schema_root
 
   def create_schema(name, json)
+    JSON::Validator.clear_cache
```
Comment (seanpdoyle) [inline spec/support/file_helpers.rb]: What do you think about doing this in a `before` hook?

```rb
RSpec.configure do |config|
  config.before(:each) { JSON::Validator.clear_cache }
end
```

Comment (anthonator) [inline spec/support/file_helpers.rb]: Sounds good.

**Comment by anthonator** [comment]
@seanpdoyle anything else you can think of before this is good to go?

**Comment by seanpdoyle** [comment]
Merged in https://github.com/thoughtbot/json_matchers/commit/a6d60ffdfe92216cd6147befe368614eb469025f. 

Thanks!

**Pull request closed without merging**